tracer/build/_build/Build.Steps.cs

```diff
@@ -226,12 +246,39 @@ bool RequiresThoroughTesting()
             })
             .ToList();
 
+        // Check if any of the integration areas have changed
+        foreach (var file in gitChangedFiles)
+        {
+            foreach (var area in integrationAreaChanges.Keys.ToArray())
+            {
+                if (integrationAreaPaths[area].Any(file.Contains))
+                {
+                    integrationAreaChanges[area] = true;
+                    Logger.Information($"Integration area {area} has changed due to {file} - full test suite will be run for that area");
+                }
+            }
+        }
+
+        var changedAreas = integrationAreaChanges
+            .Where(kvp => kvp.Value)
+            .Select(kvp => kvp.Key)
+            .ToList();
+
+        // Set the environment variable with changed areas (even if we end up doing a full test)
+        // This allows tests to be selective even during a full test run if needed
+        // Note though that this can only be reached on PRs
+        if (changedAreas.Count != 0)
+        {
+            // TODO have a way to override this if some generic file is changed that affects _all_ areas
+            Environment.SetEnvironmentVariable("CHANGED_INTEGRATION_AREAS", string.Join(",", changedAreas));
+        }
+
         var hasIntegrationChanges = gitChangedFiles.Any(s => integrationChangedFiles.Any(s.Contains));
         var snapshotChangeCount = gitChangedFiles.Count(s => s.EndsWith("verified.txt"));
 
         // If the integrations have changed, we should play it safe and test all the frameworks
         // If a lot of snapshots have changed, we should play it safe
-        return hasIntegrationChanges || (snapshotChangeCount > 100);
+        return changedAreas.Count != 0 || hasIntegrationChanges || (snapshotChangeCount > 100);
```

Comment:
if you decide to declare `areasChanged` above, you would need to change it here too

```suggestion
        return areasChanged || hasIntegrationChanges || (snapshotChangeCount > 100);
```